airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partition_enqueuer.py

```diff
@@ -27,11 +27,16 @@ def generate_partitions(self, partition_generator: PartitionGenerator, sync_mode
         Generate partitions from a partition generator and put them in a queue.
         When all the partitions are added to the queue, a sentinel is added to the queue to indicate that all the partitions have been generated.
 
+        If an exception is encountered, the exception will be caught and put in the queue.
+
         This method is meant to be called in a separate thread.
         :param partition_generator: The partition Generator
         :param sync_mode: The sync mode used
         :return:
         """
-        for partition in partition_generator.generate(sync_mode=sync_mode):
-            self._queue.put(partition)
-        self._queue.put(self._sentinel)
+        try:
+            for partition in partition_generator.generate(sync_mode=sync_mode):
+                self._queue.put(partition)
+            self._queue.put(self._sentinel)
+        except Exception as e:
```

Comment:
this is [executed on the threadpool](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py#L89) so we should catch all exceptions. They'll otherwise be swallowed and lost until we [verify that none of the futures failed](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/thread_based_concurrent_stream.py#L136-L138).